aten/src/ATen/native/cuda/MaxUnpooling.cu

```diff
@@ -0,0 +1,590 @@
+              grid,
+              block,
+              0,
+              at::cuda::getCurrentCUDAStream()>>>(
+              self.packed_accessor<scalar_t, 4>(),
+              indices.packed_accessor<int64_t, 4>(),
+              output_.data<scalar_t>(),
+              oT,
+              oH,
+              oW,
+              offsetZ);
+          AT_CHECK(
+              cudaGetLastError() == cudaSuccess,
+              "max_unpooling3d_forward_kernel failed with error code ",
+              cudaGetLastError());
+          totalZ -= 65535;
+          offsetZ += 65535;
+        }
+      }));
+  return output_;
+}
+
+Tensor max_unpooling3d_forward_cuda(
+    const Tensor& self,
+    const Tensor& indices,
+    IntList output_size,
+    IntList stride,
+    IntList padding) {
+  auto output = at::empty({0}, self.options());
+  max_unpooling3d_forward_out_cuda(
+      output, self, indices, output_size, stride, padding);
+  return output;
+}
+
+at::Tensor& max_unpooling2d_backward_out_cuda(
+    Tensor& grad_input,
+    const Tensor& grad_output_,
+    const Tensor& self_,
+    const Tensor& indices_,
+    IntList output_size) {
+  int64_t oheight = output_size[0];
+  int64_t owidth = output_size[1];
+  AT_CHECK(
+      indices_.scalar_type() == at::ScalarType::Long,
+      "elements in indices should be type int64");
+  TensorArg grad_input_arg{grad_input, "grad_input", 1},
+      grad_output_arg{grad_output_, "grad_output_", 2},
+      self_arg{self_, "self_", 3}, indices_arg{indices_, "indices_", 4};
+  checkAllSameGPU(
+      "max_unpooling2d_backward_out_cuda",
+      {grad_input_arg, grad_output_arg, self_arg, indices_arg});
+
+  AT_CHECK(
+      (self_.ndimension() == 3 || self_.ndimension() == 4),
+      "Input to max_unpooling2d should be a 3d or 4d Tensor, instead got: ",
+      self_);
+
+  AT_CHECK(
+      self_.sizes() == indices_.sizes(),
+      "Input should have same shape as indices");
+
+  AT_CHECK(output_size.size() == 2, "output_size must have two elements");
+
+  int64_t nInputCols, nInputRows, nInputPlane, batchSize;
+
+  int dimw = 2;
+  int dimh = 1;
+
+  auto self = self_.contiguous();
+  auto indices = indices_.contiguous();
+  auto grad_output = grad_output_.contiguous();
+
+  if (self.ndimension() == 3) {
+    nInputPlane = self.size(0);
+    batchSize = 1;
+  } else {
+    ++dimw;
+    ++dimh;
+    nInputPlane = self.size(1);
+    batchSize = self.size(0);
+  }
+
+  nInputCols = self.size(dimw);
+  nInputRows = self.size(dimh);
+
+  if (oheight != grad_output.size(dimh) || owidth != grad_output.size(dimw)) {
+    AT_ERROR(
+        "Inconsistent gradOutput size. output height: ",
+        oheight,
+        ", output width= ",
+        owidth,
+        ", gradOutput: ",
+        grad_output.size(dimh),
+        "x",
+        grad_output.size(dimw));
+  }
+
+  grad_input.resize_as_(self);
+  grad_input.zero_();
+
+  int count = self.numel();
+
+  dim3 block(512);
+  dim3 grid((count + 512 - 1) / 512);
```

Comment:
Err... were you planning to restore `GET_BLOCKS` and `CUDA_NUM_THREADS` use here?